lib/go-atscfg/servercachedotconfig.go

```diff
@@ -48,15 +48,15 @@ func makeCacheDotConfigMid(
 
 	dses := map[tc.DeliveryServiceName]serverCacheConfigDS{}
 	for _, ds := range deliveryServices {
-		if ds.XMLID == nil || ds.Active == nil || ds.OrgServerFQDN == nil || ds.Type == nil {
+		if &ds.XMLID == nil || &ds.Active == nil || ds.OrgServerFQDN == nil || ds.Type == nil {
```

Comment:
Instead of checking `&ds.XMLID == nil`, check `ds.XMLID == ""`.

Reply:
done